libcaja-private/caja-file-operations.c

```diff
@@ -955,7 +955,7 @@ init_common (gsize job_size,
 	CommonJob *common;
 	GdkScreen *screen;
 
-	common = g_malloc0 (job_size);
+	common = g_malloc0 (job_size * sizeof (CommonJob));
```

Comment:
Don't do that, you'll be allocating an unreasonable amount of memory.

The warning is telling you that you're allocating something not a multiple of `sizeof(CommonJob)`, which is true, but also which is *what you want*.  `CommonJob` is the "base class" for all jobs, and this allocation allocates this, plus whatever job-specific extra there is.  Such scheme allows to consider every jobs being `CommonJob`s, as they have the same memory layout for the first `sizeof(CommonJob)` bytes, but after this they have specific extra memory.

This is a typical "check you're doing it right" type of warning, not a "there *definitely* is a *bug* in your program" one.
I'm not sure what you can do if you just wanna hide this warning that is a *false positive*, maybe cast to `void*` or something.  But don't just allocate a lot more memory just to silence this: it means multiplying by about 92 on amd64, which means that for example the smallest job, `EmptyTrashJob` will go from using about 120 bytes to about 11040 bytes (!!). Similarly, a bigger job like `CreateJob` which takes about 180 bytes now would then take 16560 (16+kb) bytes.

Reply:
That is entirely unacceptable. Better to simply add a comment to the existing code referring to the clang warning and stating that this is the intended behavior, to allocate only the amount of RAM actually needed. Bloating up software's RAM consumption, when a whole OS does this, contributes to the "page's law" mess, in which new computers with new software installed run no faster (or in this case have no more extra RAM), than old ones with older, lighter software.

Reply:
@cwendling thanks for the info, I like to silence the warning, I tried, but I don't know how to do that